pkg/controller/jobframework/reconciler.go

```diff
@@ -442,8 +442,20 @@ func (r *JobReconciler) ensureOneWorkload(ctx context.Context, job GenericJob, o
 	log := ctrl.LoggerFrom(ctx)
 
 	if usePrebuiltWorkload, prebuiltWorkloadName := prebuiltWorkload(job); usePrebuiltWorkload {
-		wl, getErr := r.getPrebuiltWorkload(ctx, job, object, prebuiltWorkloadName)
-		return wl, getErr
+		wl := &kueue.Workload{}
+		err := r.client.Get(ctx, types.NamespacedName{Name: prebuiltWorkloadName, Namespace: object.GetNamespace()}, wl)
+		if err != nil {
+			return nil, client.IgnoreNotFound(err)
```

Comment:
Why do we wrap the error with NotFound? The reason could be different, for example a timeout from API server.

Reply:
We don't want to retry in case of not found.

Reply:
In that case we should check if this is NotFound error with `apierrors.IsNotFound`, instead of wrapping any error unconditionally. Example: https://github.com/kubernetes/kubernetes/blob/0c645922edcc06adff43c70c02fb56751364bbb5/pkg/controller/job/job_controller.go#L630

Reply:
`IgnoreNotFound ` is doing exactly that

Reply:
Oh I see, thanks for clarifying